src/common/tracer.h

```diff
@@ -1,32 +1,95 @@
 // -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
-#ifndef TRACER_H_
-#define TRACER_H_
+#ifndef TRACER_H
+#define TRACER_H
+
+#include "acconfig.h"
+
+#ifdef HAVE_JAEGER
 
 #define SIGNED_RIGHT_SHIFT_IS 1
 #define ARITHMETIC_RIGHT_SHIFT 1
-
 #include <jaegertracing/Tracer.h>
 
 typedef std::unique_ptr<opentracing::Span> jspan;
+namespace tracing {
+
+class Tracer {
+ private:
+  const static std::shared_ptr<opentracing::Tracer> noop_tracer;
+  std::shared_ptr<opentracing::Tracer> open_tracer;
+
+ public:
+  Tracer() = default;
+  Tracer(jaegertracing::Config& conf);
+  Tracer(opentracing::string_view service_name);
+
+  void init(opentracing::string_view service_name);
```

Comment:
we probably don't need the "init()" function, it is invoked only from one of the constructors